airflow-core/docs/public-airflow-interface.rst

```diff
@@ -77,64 +152,86 @@ You can read more about dags in :doc:`Dags <core-concepts/dags>`.
 
 References for the modules used in dags are here:
 
-.. toctree::
-  :includehidden:
-  :glob:
-  :maxdepth: 1
-
-  _api/airflow/models/dag/index
-  _api/airflow/models/dagbag/index
+.. note::
+   The airflow.sdk namespace provides the primary interface for DAG authors in Airflow 3.0+.
+   For detailed API documentation, see the `Task SDK Reference <https://airflow.apache.org/docs/task-sdk/stable/>`_.
 
-Properties of a :class:`~airflow.models.dagrun.DagRun` can also be referenced in things like :ref:`Templates <templates-ref>`.
-
-.. toctree::
-  :includehidden:
-  :glob:
-  :maxdepth: 1
+.. note::
+   The :class:`~airflow.models.dagbag.DagBag` class is used internally by Airflow for loading DAGs
+   from files and folders. DAG authors should use the :class:`~airflow.sdk.DAG` class from the
+   airflow.sdk namespace instead.
 
-  _api/airflow/models/dagrun/index
+.. note::
+   The :class:`~airflow.models.dagrun.DagRun` class is used internally by Airflow for DAG run
+   management. DAG authors should access DAG run information through the Task Context via
+   :func:`~airflow.sdk.get_current_context` or use the :class:`~airflow.sdk.types.DagRunProtocol`
+   interface.
 
 .. _pythonapi:operators:
 
 Operators
----------
+=========
 
-The base classes :class:`~airflow.models.baseoperator.BaseOperator` and :class:`~airflow.sensors.base.BaseSensorOperator` are public and may be extended to make new operators.
+The base classes :class:`~airflow.sdk.BaseOperator` and :class:`~airflow.sdk.BaseSensorOperator` are public and may be extended to make new operators.
+
+The recommended base class for new operators is :class:`~airflow.sdk.BaseOperator`
+from the airflow.sdk namespace.
 
 Subclasses of BaseOperator which are published in Apache Airflow are public in *behavior* but not in *structure*.  That is to say, the Operator's parameters and behavior is governed by semver but the methods are subject to change at any time.
 
 Task Instances
---------------
+==============
 
 Task instances are the individual runs of a single task in a DAG (in a DAG Run). They are available in the context
-passed to the execute method of the operators via the :class:`~airflow.models.taskinstance.TaskInstance` class.
+passed to the execute method of the operators via the :class:`~airflow.sdk.types.RuntimeTaskInstanceProtocol` class.
 
-.. toctree::
-  :includehidden:
-  :glob:
-  :maxdepth: 1
-
-  _api/airflow/models/taskinstance/index
+In Airflow 3.0+, task instances are accessed through the Task Context via :func:`~airflow.sdk.get_current_context`
```

Comment:
Don't add versions in docs please. Remove all such occurrences

Reply:
Removed it